src/components/ArticleList.jsx

```diff
@@ -0,0 +1,51 @@
+import React, { useEffect, useState } from "react";
+import { getArticles } from "../api.js";
+import ArticleCard from "./ArticleCard";
+import SortFilterBar from "./SortFilterBar";
+
+const ArticleList = () => {
+  const [articles, setArticles] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [sortBy, setSortBy] = useState("created_at");
+  const [filterBy, setFilterBy] = useState("");
+  const [isError, setIsError] = useState(false);
+
+  useEffect(() => {
+    getArticles()
+      .then((data) => {
+        let filteredArticles = data.articles;
+
+        if (filterBy) {
+          filteredArticles = filteredArticles.filter(
+            (article) => article.topic === filterBy
+          );
+        }
+
+        filteredArticles.sort((a, b) => b[sortBy] - a[sortBy]);
```

Comment:
This mutates the array in place. Use toSorted() to avoid mutation.

The only reason this works okay at the moment is because you're forcing a default sort by on the component render. If you're aware of it, then maybe it's justifiable.